rfcs/convergence/authoring-stories.md

```diff
@@ -0,0 +1,290 @@
+  - [4. should we render descriptions table in Canvas Control pane](#4-should-we-render-descriptions-table-in-canvas-control-pane)
+  - [5. how to author e2e suites for those stories](#5-how-to-author-e2e-suites-for-those-stories)
+  - [6. how to properly annotate stories with TS metadata to get the best DX possible](#6-how-to-properly-annotate-stories-with-ts-metadata-to-get-the-best-dx-possible)
+  - [7. how structure stories in the storybook nav tree](#7-how-structure-stories-in-the-storybook-nav-tree)
+  - [8. dissecting big story files into smaller ones](#8-dissecting-big-story-files-into-smaller-ones)
+  - [Pros and Cons](#pros-and-cons)
+- [Discarded Solutions](#discarded-solutions)
+- [Open Issues](#open-issues)
+
+<!-- tocstop -->
+
+## Summary
+
+Currently we have no style-guide/functionality requirements on how to be consistent when writing stories for converged components. This RFC should describe (as we progress) the style we want going forward.
+
+## Problem statement
+
+For convergence (vNext) we agreed on a collocated stories approach that uses [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf) for implementation.
+
+**What are we missing:**
+
+1. what should we use for interactive props playground - controls or knobs?
+2. how to use `argTypes`
+3. should controls work for all stories or only for general/default one
+4. should we render descriptions table in Canvas Control pane
+5. how to author e2e suites for those stories
+6. how to properly annotate stories with TS metadata to get the best DX possible
+7. how structure stories in the storybook nav tree / @miroslavstastny
+8. what to do if story files are getting way too big (dissecting big story files into smaller ones) / @ling1726
+
+> **💡 NOTE:**
+> This RFC will address the first 4 points for now (should be updated with others later).
+
+## Detailed Design or Proposal
+
+### 1. what should we use for interactive props playground - controls or knobs?
+
+**Proposal:**
+
+We should use [controls](https://storybook.js.org/docs/react/essentials/controls).
+
+**Why:**
+
+- while there are some limitations with using controls (they don't mirror the state in URL thus cannot be tested in e2e scenarios), using them is an industry standard in the open source world, also knobs will be deprecated in the next major storybook versions.
+
+### 2. how to use `argTypes`
+
+A pattern that is used in some of the converged stories is to define everything by hand:
+
+```ts
+AccordionExample.argTypes = {
+  inline: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  navigable: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  circular: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  multiple: {
+    defaultValue: false,
+    control: 'boolean',
+  },
+  // ... other definitions of controls
+};
+```
+
+**Proposal:**
+
+- this is not necessary as storybook generates all those controls automatically from TS metadata (Props interface)
```

Comment:
NOTE: we still need to manually provide those until all vNext and react-components will be migrated to new DX https://github.com/microsoft/fluentui/issues/18514